### DISCUSSION TITLE
Subscribing messagebroker like redis pub/sub in python socketio

### DISCUSSION THREAD

**Discussion by farhan888** [miguelgrinberg/python-socketio#1286] [Q&A] [answered]
I want to use a message broker to emit messages from the socketio. Usually I do it using the following code:

```
 async def background_task():
    while True:
        await sio.sleep(1)
        keys = redis_client.keys('*')
            
        for key in keys:   
                data = redis_client.get(key)
                await sio.emit('result', data)
        await sio.emit('my_response', {'data': 'Server generated event'})

```
However, when I remove the `await sio.sleep(1)`, socket stops taking new connections as it continuously keeps doing the background task. P.S. the background task is triggered with `sio.start_background_tast()` method. I think because of the same reason if I bring in a message broker listener(subscriber for redis pub/sub), the socket only does background task infinitely and doesn't accept any new connections. What is the best way to overcome this?

**Comment by miguelgrinberg** [comment, marked as answer]
You are using the blocking redis client. Since this is an async application you should use the async redis client to prevent blocking.

**Reply by farhan888** [reply]
Thanks a lot.  AsyncRedis solved the problem.
